src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -409,7 +420,8 @@ private void RetrieveMediaEntityStreamPaths(IEdmEntityType entityType, ODataPath
                 // ~/entityset/{key}/single-valued-Nav/subtype
                 CreateTypeCastPaths(currentPath, convertSettings, navigationProperty.DeclaringType, navigationProperty, targetsMany);
 
-                if (navigation?.Referenceable == true)
+                if (navSourceRestrictionType?.Referenceable == true ||
+                    navPropRestrictionType?.Referenceable == true)
```

Comment:
```suggestion
                    navPropRestrictionType?.Referenceable ?? false)
```

Reply:
This suggestion throws the error:

![image](https://user-images.githubusercontent.com/40403681/155354309-d9536f2a-49ea-4232-adb5-a23c075fea91.png)

Reply:
so maybe this instead

```CSharp
if ((navSourceRestrictionType?.Referenceable ?? false) ||
                    (navPropRestrictionType?.Referenceable ?? false))
```

Reply:
That works